test/dummies/accounts.ts

```diff
@@ -1,71 +1,62 @@
+    },
+    privateKey: toBase64(privkey),
+    mnemonic
+  };
+}
 
-export const aliceEncodedMnemonic =
-  '{"type":"directsecp256k1hdwallet-v1","kdf":{"algorithm":"argon2id","params":{"outputLength":32,"opsLimit":24,"memLimitKib":12288}},"encryption":{"algorithm":"xchacha20poly1305-ietf"},"data":"s2djbGL0yqGBt05QoVRb2lHxJcABnIQowMF94HujK3U0GQiu/G0Dfm4qPGi8Wp+b5TkCvJIHkaWDeFNDLsnRw9++tvkFV4LafwL3Kxp7C6ihxTQfO0vVwvofQxZDTDspxCVX5hSkUzrxey+xeeuU8KbxPB/jbetG0kCOHr9SjRlSXfMmX9Tr2/K7Xi2ZBgxJxYxFryItvyPRWYOlAUEYN8L8FLeV/2RdXNidgDjps+uOypHsh5W1ZPGmh27tonuMG2OX7Com9wOXsKgtk9FoHqDA0B49OMMjCeNAGe1j4pUhIHw6P1ig0szFrw5POb5RtqnwcevJoHgym1UiJbkf93QUEc/5cVI7+yXU"}';
-export const bobEncodedMnemonic =
-  '{"type":"directsecp256k1hdwallet-v1","kdf":{"algorithm":"argon2id","params":{"outputLength":32,"opsLimit":24,"memLimitKib":12288}},"encryption":{"algorithm":"xchacha20poly1305-ietf"},"data":"jhxF+tuyTuOVMTuR65fftyTonbzbbitysukRM5Qli2ehFGm7gKsjPDzWB7249na93/tFXmD5utaUlfQ6hP6IgAph9YadXbIyH6u10WiGqxP9zMDPVxUPElJ6rQt+f8WjgADAoBhxgzLcjCCMNQK1yoEwiWW3xSAi58QFD492Xh9i7FUPG7DzeSWEwog6ROwmpbW2LixEmLD8EW/T8QgIEXTUs34sI9l0zyO4sYtPN/01V1gcl2BQOS+zFC1OwAIfvSDelkWOfyeKYNaESof5Hw4jWjVzzsGsySVWvABj3qeuPqLLLywkEhjfCRGaSZAd/+6VEqhnsKNXl6j4ItzeOuBC5MpTC2SKcw=="}';
-
-export const alicePublicKey = {
-  '@type': 'secp256k1',
-  key: 'AhVdMs9/rOISUT9vCep8T0Se/JED1Lwy/XADdoKvN253',
-};
-export const bobPublicKey = {
-  '@type': 'secp256k1',
-  key: 'Az2CRFc4sD2ITRx9T86SmzcHHvd6u32gV795sUUvYk4q',
-};
-
-export const aliceAccountType = AccountType.LOCAL;
-export const bobAccountType = AccountType.LOCAL;
-
-export const aliceAccountBase = {
-  name: aliceAccountName,
-  address: aliceAddress,
+export const aliceAccountName = 'Alice';
+export const aliceAccountInstance = createTestAccount(aliceAccountName);
+export const {
   type: aliceAccountType,
-};
-export const bobAccountBase = {
-  name: bobAccountName,
-  address: bobAddress,
-  type: bobAccountType,
-};
-
-export const aliceAccountInstance = {
-  ...aliceAccountBase,
+  address: aliceAddress,
   publicKey: alicePublicKey,
-};
-export const bobAccountInstance = {
-  ...bobAccountBase,
-  publicKey: bobPublicKey,
-};
+  mnemonic: aliceMnemonic,
+} = aliceAccountInstance;
 
-export const aliceAccountWithMnemonic = {
-  ...aliceAccountInstance,
-  mnemonic: aliceEncodedMnemonic,
-};
-export const bobAccountWithMnemonic = {
-  ...bobAccountInstance,
-  mnemonic: bobEncodedMnemonic,
-};
+export const aliceStoreEntry = toEntryTag(aliceAccountInstance);
+export const aliceStoredAccount = JSON.stringify(aliceAccountInstance);
 
-export const aliceStoredAccount = JSON.stringify(aliceAccountWithMnemonic);
-export const bobStoredAccount = JSON.stringify(bobAccountWithMnemonic);
+export const bobAccountName = 'Bob';
+export const bobAccountInstance = createTestAccount(bobAccountName);
+export const {
+  type: bobAccountType,
+  address: bobAddress,
+  publicKey: bobPublicKey,
+  mnemonic: bobMnemonic,
+} = bobAccountInstance;
```

Comment:
This notation with naming the export at the end is new to me, does it have any special purpose/functionality?

Reply:
This replaces the need to export every single variable separately. Both `Alice` and `Bob` used the same object structures, so it was cleaner to create a factory method and fetch the data using a [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

Reply:
Cool! I didn't know destructuring could be used this way for exports, thanks for the explanation!